### PR TITLE
Image registry and setup update

### PR DESCRIPTION
**Pull request by georglauterbach**
# Description

Follow up to #3219.

Changes `docker.io/mailserver` for `ghcr.io/docker-mailserver`. Therefore updates `setup.sh` as well. So I figured it's okay to slightly adjust the docs for `setup.sh` as well.

## Type of change

- [x] Update (non-breaking change that does improve existing functionality)
- [x] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (README.md or the documentation under `docs/`)
- [x] If necessary I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: docs/content/config/setup.sh.md [hunk 1]
```diff
@@ -1,123 +1,26 @@
...
 
-The latest version of the script is included in the `docker-mailserver` repository. You may retrieve it at any time by running this command in your console:
+    `setup.sh` is not required since v10.2.0. We encourage you to use `docker exec -ti <CONTAINER NAME> setup` instead.
 
-```sh
-wget https://raw.githubusercontent.com/docker-mailserver/docker-mailserver/master/setup.sh
-chmod a+x ./setup.sh
-```
-
-## Usage
```
Comment (georglauterbach) [inline docs/content/config/setup.sh.md:16]: I always thought this to be rather useless. I hope you agree :)

File: docs/content/config/setup.sh.md [hunk 2]
```diff
@@ -1,123 +1,26 @@
...
+!!! warning
 
-    COMMAND quota :=
-        ./setup.sh quota set <EMAIL ADDRESS> [<QUOTA>]
-        ./setup.sh quota del <EMAIL ADDRESS>
+    This script assumes Docker or Podman is used. You will not be able to use `setup.sh` with other container orchestration tools.
 
-    COMMAND config :=
-        ./setup.sh config dkim [ ARGUMENTS... ]
+[`setup.sh`][github-file-setupsh] is a script that aids in running commands inside your DMS container, including initial configuration. It has become a wrapper around `docker exec -ti <CONTAINER NAME> setup` where it basically tries to automatically determine `<CONTAINER NAME>`.
```
Comment (polarathene) [inline docs/content/config/setup.sh.md:15]:
```suggestion
[`setup.sh`][github-file-setupsh] is a script that is complimentary to the internal `setup` command in `docker-mailserver`.

It mostly provides the convenience of aliasing `docker exec -ti <CONTAINER NAME> setup`, inferring the container name of a running `docker-mailserver` instance or running a new instance and bind mounting necessary volumes implicitly.
```

File: docs/content/config/setup.sh.md [hunk 3]
```diff
@@ -1,123 +1,26 @@
...
-    message is shown describing the error. In case of an error, the script will exit with exit
-    status 1.
+It is intended to be run from the host machine, _not_ from inside your running container. The latest version of the script is included in the `docker-mailserver` repository. You may retrieve it at any time by running this command in your console:
 
+```sh
+wget https://raw.githubusercontent.com/docker-mailserver/docker-mailserver/master/setup.sh
+chmod a+x ./setup.sh
 ```
 
+By running `./setup.sh help`, usage information is printed.
```
Comment (polarathene) [inline docs/content/config/setup.sh.md:24]:
```suggestion
For more information on using the script run: `./setup.sh help`.
```

**Review by polarathene — CHANGES_REQUESTED** [review]

File: target/bin/setup [hunk 4]
```diff
@@ -13,89 +11,80 @@ function _usage
...
+    setup - 'docker-mailserver' Administration & Configuration CLI
 
 ${ORANGE}SYNOPSIS${RESET}
-    ./${SCRIPT:-${0}} [ OPTIONS${RED}...${RESET} ] COMMAND [ help ${RED}|${RESET} ARGUMENTS${RED}...${RESET} ]
+    setup [ OPTIONS${RED}...${RESET} ] COMMAND [ help ${RED}|${RESET} ARGUMENTS${RED}...${RESET} ]
 
     COMMAND ${RED}:=${RESET} { email ${RED}|${RESET} alias ${RED}|${RESET} quota ${RED}|${RESET} dovecot-master ${RED}|${RESET} config ${RED}|${RESET} relay ${RED}|${RESET} debug } SUBCOMMAND
 
 ${ORANGE}DESCRIPTION${RESET}
-    This is the main administration script that you use for all your interactions with
```
Comment (polarathene) [inline target/bin/setup:24]: If you want to keep some of this description that was removed, you would want to relocate it to `setup.sh` in some manner, or the docs on `setup.sh`. I don't think it's relevant anymore to be interleaved into the main `setup` command help output.

---

I was not able to provide a web UI suggestion for changes and edited the file myself. The commit message detais:

- Remove dynamic support for showing command invoker via `${0}` for `./setup.sh` or `/usr/local/bin/setup`, just show `setup` command.
- Remove bulk of description text that isn't applicable to the `setup` command, but `./setup.sh` only.
- Revised description for help usage, as attempting `setup email help` or similar is not valid. Clarify to user they must be more specific.
- Revised the 2nd example description.
- Changed references of "script" to "CLI" which is now more accurate representation of the `setup` command.

Comment (polarathene) [reply target/bin/setup:24]: Note that most commands call scripts that will still output `./setup.sh` in their `help` output.

File: docs/content/config/setup.sh.md [hunk 5]
```diff
@@ -1,123 +1,26 @@
...
+title: About setup.sh
 hide:
   - toc
 ---
 
-[`setup.sh`][github-file-setupsh] is an administration script that helps with the most common tasks, including initial configuration. It is intended to be run from the host machine, _not_ from inside your running container.
+!!! note
 
-The latest version of the script is included in the `docker-mailserver` repository. You may retrieve it at any time by running this command in your console:
+    `setup.sh` is not required since v10.2.0. We encourage you to use `docker exec -ti <CONTAINER NAME> setup` instead.
```
Comment (polarathene) [inline docs/content/config/setup.sh.md:9]: If you want to keep this, that's fine.

I have made commits to your PRs which removed this admonition from other doc pages however. Changelog for v12 could probably express the information just as well.

---

This is clarified further down, so mostly redundant. If you'd like to keep it, maybe drop the version mention?

```suggestion
    `setup.sh` is not required. We encourage you to use `docker exec -ti <CONTAINER NAME> setup` instead.
```

**Review by polarathene — CHANGES_REQUESTED** [review]

**Review by polarathene — APPROVED** [review]

**Review by wernerfred — APPROVED** [review]

**Review by casperklein — APPROVED** [review]

**Pull request merged**